lib/iris/cube.py

```diff
@@ -758,6 +767,231 @@ def _is_single_item(testee):
     return isinstance(testee, str) or not isinstance(testee, Iterable)
 
 
+class CubeAttrsDict(MutableMapping):
+    """
+    A dict-like object for "Cube.attributes", which provides unified user access to
+    the combined cube 'local' and 'global' attributes, mimicking the behaviour of a
+    simple dict.
+
+    This supports all the regular methods of a 'dict',
+    However, a few things such as the detailed print (repr) are different.
+
+    In addition, the 'locals' and 'globals' properties provide specific access to local
+    and global attributes, as regular :class:`~iris.common.mixin.LimitedAttributeDict`s.
+
+    Notes
+    -----
+    For type testing, "issubclass(CubeAttrsDict, Mapping)" is True, but
+    "issubclass(CubeAttrsDict, dict)" is False.
+
+    Properties 'locals' and 'globals' are the two sets of cube attributes.  These are
+    both :class:`~iris.common.mixin.LimitedAttributeDict`.  The CubeAttrsDict object
+    contains *no* additional state of its own, but simply acts as a view on these two.
+
+    When reading (__getitem__, pop. popitem, keys, values etc), it contains all the
+    keys + values of both 'locals' and 'globals'.  When a key occurs in *both* 'locals'
+    and 'globals', the result is the local value.
+
+    When writing (__setitem__, setdefault, update, etc) to a key already present, the
+    existing entry in either 'locals' or 'globals' is updated.  If both are present,
+    'locals' is updated.
```

Comment:
Why did you choose not to update both?
I just wonder whether this may catch some users out.e

Reply:
It may be worth drawing more attention to this in the docstring. Currently, you mention it https://github.com/SciTools/iris/blob/7db42ce754e1d6d8b22f69aed2baa3eb4406c250/lib/iris/cube.py#L795-L797 but I wonder if including example, e.g. 
```
>>> a = CubeAttrsDict(locals={"z": "this"}, globals={"z": "that"})
>>> a['z'] = 'other'
>>> a
CubeAttrsDict(globals={'z': 'that'}, locals={'z': 'other'})
>>> a.globals['z'] = 'else'
>>> a
CubeAttrsDict(globals={'z': 'else'}, locals={'z': 'other'})
```
or even just highlighting this sentence may help?

Reply:
I'm struggling with this suggestion, as I think it would over-emphasize the really _rare_ case when an attribute is found in both locals + globals.  
To me, it's much more natural to start by saying that any _existing_ attribute will be updated.  So, saying "locals is the default" is really unhelpful + will give the wrong impression, in my view.